zfsbootmenu/bin/zsnapshots

```diff
@@ -19,18 +20,61 @@ done
 
 unset src sources
 
+list_root_filesystems() {
+  local fs mountpoint active ret
+
+  ret=1
+  while IFS=$'\t' read -r fs mountpoint active; do
+    if [ "${mountpoint}" = "/" ]; then
+      [ "${active}" = "off" ] && continue;
+    elif [ "${mountpoint}" = "legacy" ]; then
+      [ "${active}" = "on" ] || continue;
+    else
+      continue;
+    fi
+
+    echo "${fs}"
+    ret=0
+  done <<< "$( zfs list -H -t filesystem -o name,mountpoint,org.zfsbootmenu:active )"
+  return "${ret}"
+}
+
 # Replace the global_header function with a stub
 global_header() {
   echo -n -e "\\033[1;33m[ Recover from snapshot ]"
 }
 
-if [ $# -ne 1 ] ; then
- echo "Usage: $0 filesystem"
- exit
-fi
+fs_header() {
+  echo -n -e "\\033[1;33m[ Select a filesystem ]"
+}
 
 fs="${1}"
 
+if [ -z "${fs}" ]; then
+  if ! candidates="$(list_root_filesystems 2>/dev/null)"; then
```

Comment:
The new function should be used in place of the one-off `list_root_filesystems`

Reply:
I can't believe I forgot to change this!